zerver/tests/test_message_edit.py

```diff
@@ -1893,6 +1893,149 @@ def assert_has_visibility_policy(
         assert_has_visibility_policy(cordelia, target_topic, UserTopic.VisibilityPolicy.UNMUTED)
         assert_has_visibility_policy(aaron, target_topic, UserTopic.VisibilityPolicy.UNMUTED)
 
+    def test_user_topic_states_on_moving_to_topic_with_no_messages(self) -> None:
+        stream_name = "Stream 123"
+        stream = self.make_stream(stream_name)
+
+        hamlet = self.example_user("hamlet")
+        cordelia = self.example_user("cordelia")
+        aaron = self.example_user("aaron")
+
+        self.subscribe(hamlet, stream_name)
+        self.subscribe(cordelia, stream_name)
+        self.subscribe(aaron, stream_name)
+
+        def assert_has_visibility_policy(
```

Comment:
We might want to consider making this a function not defined in 3 different individual tests